internal/controllers/reconciliation/controller.go

```diff
@@ -115,58 +115,23 @@ func (c *Controller) Reconcile(ctx context.Context, req reconstitution.Request)
 	ctx = logr.NewContext(ctx, logger)
 
 	// Find the current and (optionally) previous desired states in the cache
-	synRef := reconstitution.NewSynthesisRef(comp)
-	resource, exists := c.resourceClient.Get(ctx, synRef, &req.Resource)
+	var prev *resource.Resource
+	resource, visible, exists := c.resourceClient.Get(ctx, comp.Status.GetCurrentSynthesisUUID(), req.Resource)
 	if !exists {
 		// It's possible for the cache to be empty because a manifest for this resource no longer exists at the requested composition generation.
 		// Dropping the work item is safe since filling the new version will generate a new queue message.
```

Comment:
Unrelated to this PR, but what does "filling the new version" mean here?

Reply:
Yeahh that comment is kind of unclear. IIRC the idea is that it's safe to drop the work item because filling the cache will always generate another work item anyway

Reply:
Updated the comment for clarity